src/main/java/com/onelogin/sdk/conn/Client.java

```diff
@@ -3779,6 +3874,328 @@ public Mapping createMapping(Map<String, Object> mappingParams) throws OAuthSyst
+	 * @throws URISyntaxException - if there is an error when generating the target URL at the URIBuilder constructor
+	 * @throws ErrorResourceInitialization
+	 *
+	 * @see <a target="_blank" href="https://developers.onelogin.com/api-docs/2/user-mappings/list-condition-values">List Conditions Values documentation</a>
+	 */
+	public List<HashMap<String, String>> getMappingConditionValues(String conditionValue) throws OAuthSystemException, OAuthProblemException, URISyntaxException, ErrorResourceInitialization {
+		int versionId = settings.getVersionId("GET_MAPPING_CONDITION_VALUES_URL");
+		URIBuilder url = new URIBuilder(settings.getURL(Constants.GET_MAPPING_CONDITION_VALUES_URL, conditionValue, versionId));
+
+		return retrieveListNameValue(url, versionId);
+	}
+
+	/**
+	 * Gets a list of the actions that can be applied when a mapping runs.
+	 *
+	 * @return List<HashMap<String,String>>
+	 *
+	 * @throws OAuthSystemException - if there is a IOException reading parameters of the httpURLConnection
+	 * @throws OAuthProblemException - if there are errors validating the OneloginOAuthJSONResourceResponse and throwOAuthProblemException is enabled
+	 * @throws URISyntaxException - if there is an error when generating the target URL at the URIBuilder constructor
+	 * @throws ErrorResourceInitialization
+	 *
+	 * @see <a target="_blank" href="https://developers.onelogin.com/api-docs/2/user-mappings/list-actions">List Actions documentation</a>
+	 */
+	public List<HashMap<String, String>> getMappingActions() throws OAuthSystemException, OAuthProblemException, URISyntaxException, ErrorResourceInitialization {
+		int versionId = settings.getVersionId("GET_MAPPING_ACTIONS_URL");
+		URIBuilder url = new URIBuilder(settings.getURL(Constants.GET_MAPPING_ACTIONS_URL, versionId));
+
+		return retrieveListNameValue(url, versionId);
+	}
+
+	/**
+	 * Update order of mappings
+	 *
+	 * @param mappingIds ordered list of Mapping IDs
+	 *
+	 * @return List<Long>
+	 *
+	 * @throws OAuthSystemException - if there is a IOException reading parameters of the httpURLConnection
+	 * @throws OAuthProblemException - if there are errors validating the OneloginOAuthJSONResourceResponse and throwOAuthProblemException is enabled
+	 * @throws URISyntaxException - if there is an error when generating the target URL at the URIBuilder constructor
+	 * @throws ErrorResourceInitialization
+	 *
+	 * @see <a target="_blank" href="https://developers.onelogin.com/api-docs/2/user-mappings/bulk-sort">Bulk Sort documentation</a>
+	 */
+	public List<Long> mappingSort(List<Long> mappingIds) throws OAuthSystemException, OAuthProblemException, URISyntaxException, ErrorResourceInitialization {
+
+		int versionId = settings.getVersionId("MAPPING_SORT_URL");
+		URIBuilder url = new URIBuilder(settings.getURL(Constants.MAPPING_SORT_URL, versionId));
+
+		return setOperation(mappingIds, url, versionId);
```

Comment:
This fails, as the response is an array of ids, and not an array of objects with an id field. Not sure what the best option is fore changing this.

Reply:
Ideally extend the setOperation method to cover such use case as well

Check how was solved in python:
https://github.com/onelogin/onelogin-python-sdk/blob/v2/src/onelogin/api/client.py#L268


https://github.com/onelogin/onelogin-python-sdk/blob/v2/src/onelogin/api/util/response_handlers.py#L57

Reply:
Updated to handle the fact that responses might not be JSON objects for getting the ID.